pkg/server/commands/reverseexpand/reverse_expand.go

```diff
@@ -580,3 +606,30 @@ func (c *ReverseExpandQuery) trySendCandidate(ctx context.Context, intersectionO
 
 	return nil
 }
+
+func (c *ReverseExpandQuery) throttle(ctx context.Context, metadata *ResolutionMetadata) {
+	ctx, span := tracer.Start(ctx, "ResolveCheck")
```

Comment:
This code looks awfully similar to 

<img width="1553" alt="image" src="https://github.com/openfga/openfga/assets/5374887/3ae29cdd-e75d-40b9-bb62-ddbcefd41c85">

are we missing an abstraction here?

Reply:
Made this common 👍